### ISSUE TITLE
expo-notifications (v0.3.3) not compatible with Xcode 12

### ISSUE THREAD

**Issue by magiskaoskar** [expo/expo#10257]
### Summary of Issue
Getting error `Non-void function does not return a value in all control paths` in `EXNotifications` after upgrading Xcode to version 12.

### Environment - output of `expo diagnostics` & the platform(s) you're targeting
  Expo CLI 3.27.6 environment info:
    System:
      OS: macOS 10.15.4
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 10.21.0 - ~/.nvm/versions/node/v10.21.0/bin/node
      Yarn: 1.19.0 - /usr/local/bin/yarn
      npm: 6.14.7 - ~/.nvm/versions/node/v10.21.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.9.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.0, DriverKit 19.0, macOS 10.15, tvOS 14.0, watchOS 7.0
    IDEs:
      Android Studio: 4.0 AI-193.6911.18.40.6626763
      Xcode: 12.0/12A7209 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^38.0.0 => 38.0.10 
      react: 16.11.0 => 16.11.0 
      react-dom: 16.11.0 => 16.11.0 
      react-native: 0.62.2 => 0.62.2 
      react-native-web: ~0.11.7 => 0.11.7 
    npmGlobalPackages:
      expo-cli: 3.27.6
    Expo Workflow: bare
 
### Steps to Reproduce
- install latest Xcode version (12).
- `expo init --npm -t bare-minimum`.
- `expo install expo-notifications`.
- Run app.

### Expected Behavior vs Actual Behavior
Expected Behavior: Build without errors.
Actual Behavior: Can't build/run app.

**Comment by sjchmiela** [comment]
Have you tried `expo-notifications` version higher or equal to `0.4.0`? This is when https://github.com/expo/expo/pull/8938 landed, adding support for iOS 14 to `expo-notifications`.

**Comment by magiskaoskar** [comment]
> Have you tried `expo-notifications` version higher or equal to `0.4.0`? This is when #8938 landed, adding support for iOS 14 to `expo-notifications`.

Yes, I tried version `0.5.0` and that fixed the error but I'm scared to install a version that isn't compatible with the expo SDK version (not installing the package with `expo install` command).

**Comment by sjchmiela** [comment]
Hm… if I understand the issue correctly you must be running your project in bare workflow (in managed you wouldn't even start Xcode). In bare workflow the concept of an SDK is much more relaxed — you should be able to install `expo-notifications` even in latest version without any problems (given you follow changes in changelog [there are no breaking ones]). 🙂

**Comment by cryptixcoder** [comment]
I'm having the same issue, have you found the solution. I've tried updating to version 0.1.1 same error

**Comment by Entryist** [comment]
> I'm having the same issue, have you found the solution. I've tried updating to version 0.1.1 same error

Yes, update expo-notifications to 0.5.0 as @sjchmiela suggested. Fixed it for me.

**Comment by magiskaoskar** [comment]
@sjchmiela Corrent, I'm using the bare workflow. Alright, so in the bare workflow I can **always** install the latest version of an expo package without there being any compatible issue with the installed SDK version? Its a bit confusing when the docs for the [bare workflow](https://github.com/expo/expo/tree/master/packages/expo-notifications#installation-in-bare-react-native-projects) tells me to install the package with `expo install expo-notifications` (which chooses a compatible version for me).

**Comment by sjchmiela** [comment]
tl;dr — usually you should be safe upgrading to the latest version (not using `expo install`), but keep in mind if you use Expo client it may break and other unimodules may break if not upgraded too.

---

So… the answer would be… yes and no.

- If you use `expo install` you can keep using Expo client to develop your application as described in [this documentation page](https://docs.expo.io/bare/using-expo-client/#prefer--expo-install--over-).
- Sometimes (seldom) we release a breaking change to the core packages (`@unimodules/core`, `@unimodules/react-native-adapter`). For the dependencies to work and depend on proper version we also update the dependency requirement there.

  Imagine a project where there are `expo-notifications` and `expo-brightness` installed, both depending on a particular, compatible version of the foundation (let's say `1.0.0`). Now imagine we need to release a breaking change to the core packages — we bump their versions (they're now `2.0.0`) and we publish updated versions of `expo-notifications` and `expo-brightness` (depending on `2.0.0`).
  
  If you now installed latest version of **only** `expo-notifications`, `expo-brightness` would break, stop compiling (because it depended on an older version of `@unimodules/core` and since there can only be one version of native code linked, the most recent will be used).

  Using `expo install` and `expo upgrade` prevents this issue — the commands sync versions of dependencies to some well-known set (even if it sometimes isn't the most recent version) for an SDK version.

  Usually this type of situation shouldn't occur (we rarely introduce breaking changes in the `@unimodules` scope), but at the same time we wouldn't want to push users into this too easily.

**Comment by magiskaoskar** [comment]
@sjchmiela That makes it a lot clearer. Thanks a lot for your help!

**Comment by akinlekan28** [comment]
Just incase you're here and running a bare workflow and running on SDK 38, @Entryist solution worked for me
